lib/policy_machine.rb

```diff
@@ -194,22 +202,37 @@ def privileges
   #
   # TODO:  might make privilege a class of its own
   def scoped_privileges(user_or_attribute, object_or_attribute, options = {})
-    privileges_and_prohibitions = if policy_machine_storage_adapter.respond_to?(:scoped_privileges)
-      policy_machine_storage_adapter.scoped_privileges(user_or_attribute.stored_pe, object_or_attribute.stored_pe, options).map do |op|
-        operation = PM::Operation.convert_stored_pe_to_pe(op, policy_machine_storage_adapter, PM::Operation)
-        [user_or_attribute, operation, object_or_attribute]
-      end
-    else
-      operations.grep(->operation{is_privilege_ignoring_prohibitions?(user_or_attribute, operation, object_or_attribute)}) do |op|
-        [user_or_attribute, op, object_or_attribute]
+    # Duplicate so passed hash doesn't get modified
+    options = options.dup
+    user_attribute_scope = options.delete(:user_attribute_scope)
```

Comment:
if we have activesupport in scope, this and the `dup` could be replaced with `except`, whether or not we _should_ have activesupport in scope is another question, though

Reply:
This workflow is a bit weird; `:user_attribute_scope` actually _must_ be removed from the options hash in order to get the full list of prohibitions, which is then filtered down with a second call that gets the privileges within the scope. It's not ideal, but until the TODO is completed (surfacing the user attributes that provide a given privilege) then it's necessary, I think